integration_tests/src/main/python/spark_session.py

```diff
@@ -48,6 +48,8 @@ def _set_all_confs(conf):
 def reset_spark_session_conf():
     """Reset all of the configs for a given spark session."""
     _set_all_confs(_orig_conf)
+    """We should clear the cache"""
+    spark.catalog.clearCache()
```

Comment:
I do like the idea of clearing the cache after a run, but it might make some things more difficult.  Like with `test_cache_table`. I am fairly sure that it will have problems because it caches something without resetting things.  Then it will run a test that uses the cached data but when it sets it up for the CPU run, it will reset the config.  Then again for the GPU run it will call this function.  So either this is not actually clearing the cache or `test_cache_table` is failing.

Reply:
You are right. I have removed the `test_cache_table` because I didn't think it was testing anything thats not being tested with the now added methods. Correct me if I am wrong and I can add it back.